pkg/controllers/osc/osc_controller.go

```diff
@@ -86,29 +92,31 @@ func Add(
 	podCIDR string,
 	nodePortRange string) error {
 	reconciler := &Reconciler{
-		Client:                mgr.GetClient(),
-		log:                   log,
-		namespace:             namespace,
-		clusterAddress:        clusterName,
-		generator:             generator,
-		kubeconfig:            kubeconfig,
-		clusterDNSIPs:         clusterDNSIPs,
-		containerRuntime:      containerRuntime,
-		pauseImage:            pauseImage,
-		initialTaints:         initialTaints,
-		externalCloudProvider: externalCloudProvider,
-		nodeHTTPProxy:         nodeHTTPProxy,
-		nodeNoProxy:           nodeNoProxy,
-		podCIDR:               podCIDR,
-		nodePortRange:         nodePortRange,
+		log:                     log,
+		workerClient:            mgr.GetClient(),
+		Client:                  client,
+		workerClusterKubeconfig: workerClusterKubeconfig,
+		namespace:               namespace,
+		ospNamespace:            ospNamespace,
+		clusterAddress:          clusterName,
+		generator:               generator,
+		clusterDNSIPs:           clusterDNSIPs,
+		containerRuntime:        containerRuntime,
+		pauseImage:              pauseImage,
+		initialTaints:           initialTaints,
+		externalCloudProvider:   externalCloudProvider,
+		nodeHTTPProxy:           nodeHTTPProxy,
+		nodeNoProxy:             nodeNoProxy,
+		podCIDR:                 podCIDR,
+		nodePortRange:           nodePortRange,
 	}
 	log.Info("Reconciling OSC resource..")
 	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: reconciler, MaxConcurrentReconciles: workerCount})
 	if err != nil {
 		return err
 	}
 
-	if err := c.Watch(&source.Kind{Type: &clusterv1alpha1.MachineDeployment{}}, &handler.EnqueueRequestForObject{}); err != nil {
+	if err := c.Watch(&source.Kind{Type: &clusterv1alpha1.MachineDeployment{}}, &handler.EnqueueRequestForObject{}, filterMachineDeploymentPredicate()); err != nil {
```

Comment:
Probably you would need some cache injection here, but we don't need to worry about this now.

Reply:
Okay